Dynamic-Programming/FibonacciNumber.js

```diff
@@ -1,19 +1,27 @@
 /**
- * @function Fibonacci
+ * @function fibonacci
  * @description Fibonacci is the sum of previous two fibonacci numbers.
  * @param {Integer} N - The input integer
  * @return {Integer} fibonacci of N.
  * @see [Fibonacci_Numbers](https://en.wikipedia.org/wiki/Fibonacci_number)
  */
 const fibonacci = (N) => {
-  // creating array to store values
-  const memo = new Array(N + 1)
-  memo[0] = 0
-  memo[1] = 1
-  for (let i = 2; i <= N; i++) {
-    memo[i] = memo[i - 1] + memo[i - 2]
+  if (!Number.isInteger(N)) {
+    throw new TypeError('Input should be integer')
   }
-  return memo[N]
+
+  // memoize the last two numbers
+  let firstNumber = 0
+  let secondNumber = 1
+
+  for (let i = 1; i < N; i++) {
+    const sumOfNumbers = firstNumber + secondNumber
```

Comment:
Is this variable really required? You are only using it once, so you can replace it with the operation itself.

Reply:
I don't think it's possible due to the fact that ``firstNumber`` is updated the line after. Thus, the operation would not be the same anymore.

Reply:
Ah right. My mistake XD